locations/spiders/bestwestern.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+import scrapy
+import json
+import html
+from locations.dict_parser import DictParser
+
+
+def from_wikidata(name, code):
+    return {"brand": name, "brand_wikidata": code}
+
+
+def set_brand(item, brand):
+    item["brand"] = brand.get("brand")
+    item["brand_wikidata"] = brand.get("brand_wikidata")
+
+
+class BestWesternSpider(scrapy.spiders.SitemapSpider):
+    name = "bestwestern"
+    brands = [
+        from_wikidata("Best Western Premier", "Q830334"),
+        from_wikidata("Best Western Plus", "Q830334"),
+        from_wikidata("Aiden by Best Western", "Q830334"),
+        from_wikidata("Sure Hotel", "Q830334"),
+        from_wikidata("Surestay Plus", "Q830334"),
+        from_wikidata("Surestay", "Q830334"),
+        from_wikidata("Best Western", "Q830334"),
+    ]
+    allowed_domains = ["bestwestern.com"]
+    sitemap_urls = ["https://www.bestwestern.com/etc/seo/bestwestern/hotels.xml"]
+    sitemap_rules = [(r"/en_US/book/hotels-in-.*\.html", "parse_hotel")]
+    download_delay = 0.5
+
+    def parse_hotel(self, response):
+        data_hotel_details = response.xpath(
+            '//div[@id="hotel-details-info"]/@data-hoteldetails'
+        ).get()
+        if data_hotel_details:
+            json_data = json.loads(html.unescape(data_hotel_details))
+            summary = json_data["summary"]
+            my_name = summary["name"].lower()
+            my_brand = None
+            for brand in self.brands:
+                if my_name.startswith(brand["brand"].lower()):
+                    my_brand = brand
+                    break
+            if not my_brand:
+                # For the most part these are hotels not owned by BW, but part of a "collection"
+                return
+            item = DictParser.parse(summary)
+            set_brand(item, my_brand)
```

Comment:
```suggestion
            item.update(my_brand)
```

Reply:
Totally fair comment but I would like to leave as is for now as "from_wikidata()" and "set_brand()" are things I would like to see a more structural "fix" for and when I fix one occurrence I would fix all. grep "from_wikidata()" at least lets me find them readily at a later date.

Reply:
I'm having a hard time seeing the utility of these setter functions. What's the benefit over just including the data in `item_attributes` or setting them as part of the `parse()` function?

Reply:
When there are multiple brands then item_attributes does not cut it. The only benefit for now is that I can port my spiders in easily and then we can apply a more structural clean up. Refactoring method calls is far easier than unpicking a set of field assignments.

Reply:
If item_attributes doesn't cut it, then we should be setting the details in the parse function. As it stands, the setter function is obfuscating the field assignments and makes it harder to follow.

Reply:
Let me ask the question then. This PR #3667 was too much too soon. But hey now you have LD, microdata and "dict" parsers. Would you now be ready for something like:

https://github.com/alltheplaces/alltheplaces/blob/e993696dc443d0a5de4328993193ad16cc621f90/locations/brands.py

... coming in as part of this PR. I tjink brand/wikidata are central concepts. They deserve a bit of first class support which will mean less typing and I hope clearer code all round.

Reply:
My feedback there is the same as on this PR. I don't see the benefit of extracting brand and brand_wikidata into separate objects or functions. The spider parse functions should be where that happens. It's easier to read and update when it's all in the one spider python file.